app/connectors/ApplicationClient.scala

```diff
@@ -208,41 +213,68 @@ trait ApplicationClient {
   private def encodeUrlParam(str: String) = URLEncoder.encode(str, "UTF-8")
 
   def startPhase3TestByToken(launchpadInviteId: String)(implicit hc: HeaderCarrier): Future[Unit] = {
-    http.PUT(s"${url.host}${url.base}/launchpad/${encodeUrlParam(launchpadInviteId)}/markAsStarted", "").map(_ => ())
+    http.PUT(s"$apiBaseUrl/launchpad/${encodeUrlParam(launchpadInviteId)}/markAsStarted", "").map(_ => ())
   }
 
   def completePhase3TestByToken(launchpadInviteId: String)(implicit hc: HeaderCarrier): Future[Unit] = {
-    http.PUT(s"${url.host}${url.base}/launchpad/${encodeUrlParam(launchpadInviteId)}/markAsComplete", "").map(_ => ())
+    http.PUT(s"$apiBaseUrl/launchpad/${encodeUrlParam(launchpadInviteId)}/markAsComplete", "").map(_ => ())
   }
 
   def startTest(cubiksUserId: Int)(implicit hc: HeaderCarrier): Future[Unit] = {
-    http.PUT(s"${url.host}${url.base}/cubiks/$cubiksUserId/start", "").map(_ => ())
+    http.PUT(s"$apiBaseUrl/cubiks/$cubiksUserId/start", "").map(_ => ())
   }
 
   def completeTestByToken(token: UniqueIdentifier)(implicit hc: HeaderCarrier): Future[Unit] = {
-    http.PUT(s"${url.host}${url.base}/cubiks/complete-by-token/$token", "").map(_ => ())
+    http.PUT(s"$apiBaseUrl/cubiks/complete-by-token/$token", "").map(_ => ())
   }
 
   def confirmAllocation(appId: UniqueIdentifier)(implicit hc: HeaderCarrier): Future[Unit] = {
-    http.POST(s"${url.host}${url.base}/allocation-status/confirm/$appId", "").map(_ => ())
+    http.POST(s"$apiBaseUrl/allocation-status/confirm/$appId", "").map(_ => ())
+  }
+
+  def eventWithSessionsForApplicationOnly(appId: UniqueIdentifier, eventType: EventType)(implicit hc: HeaderCarrier): Future[List[Event]] = {
+    http.GET(
+      s"$apiBaseUrl/sessions/findByApplicationId?applicationId=$appId&sessionEventType=${eventType.toString}"
```

Comment:
You shouldn't pass query parameters like this - instead use the overloaded version of GET which handles these for you (including encoding them)